hyperspy/io_plugins/blockfile.py

```diff
@@ -126,168 +151,284 @@ def get_header_from_signal(signal, endianess='<'):
-                            'date': date,
-                            'time': time,
-                            'time_zone': time_zone,
-                            'notes': header['Note']},
-                "Signal": {'signal_type': "diffraction",
-                           'record_by': 'image', },
-                }
+    data = data.reshape((NY, NX, DP_SZ, DP_SZ), order="C").squeeze()
+
+    units = ["nm", "nm", "cm", "cm"]
+    names = ["y", "x", "dy", "dx"]
+    scales = [header["SY"], header["SX"], SDP, SDP]
+    date, time, time_zone = serial_date_to_ISO_format(header["Acquisition_time"])
+    metadata = {
+        "General": {
+            "original_filename": os.path.split(filename)[1],
+            "date": date,
+            "time": time,
+            "time_zone": time_zone,
+            "notes": header["Note"],
+        },
+        "Signal": {
+            "signal_type": "diffraction",
+            "record_by": "image",
+        },
+    }
     # Create the axis objects for each axis
     dim = data.ndim
     axes = [
         {
-            'size': data.shape[i],
-            'index_in_array': i,
-            'name': names[i],
-            'scale': scales[i],
-            'offset': 0.0,
-            'units': units[i], }
-        for i in range(dim)]
-
-    dictionary = {'data': data,
-                  'axes': axes,
-                  'metadata': metadata,
-                  'original_metadata': original_metadata,
-                  'mapping': mapping, }
+            "size": data.shape[i],
+            "index_in_array": i,
+            "name": names[i],
+            "scale": scales[i],
+            "offset": 0.0,
+            "units": units[i],
+        }
+        for i in range(dim)
+    ]
+
+    dictionary = {
+        "data": data,
+        "axes": axes,
+        "metadata": metadata,
+        "original_metadata": original_metadata,
+        "mapping": mapping,
+    }
 
     f.close()
-    return [dictionary, ]
+    return [
+        dictionary,
+    ]
 
 
 def file_writer(filename, signal, **kwds):
-    endianess = kwds.pop('endianess', '<')
+    """
+    Write signal to blockfile.
+
+    Parameters
+    ----------
+    file : str
+        Filename of the file to write to
+    signal : instance of hyperspy Signal2D
+        The signal to save.
+    endianess : str
+        '<' (default) or '>' determining how the bits are written to the file
+    intensity_scaling : str or 2-Tuple of float/int
+        If the signal datatype is not uint8 this argument provides intensity
+        linear scaling strategies. If 'dtype', the entire dtype range is mapped
+        to 0-255, if 'minmax' the range between the minimum and maximum intensity is
+        mapped to 0-255, if 'crop' the range between 0-255 is conserved without
+        overflow, if a tuple of values the values between this range is mapped
+        to 0-255. If None (default) no rescaling is performed and overflow is
+        permitted.
+    vbf : str or Signal2D
```

Comment:
It is better to avoid abbreviation, would `navigator` be better?

Reply:
navigator="navigator" seems a bit strange to me though. maybe `nav_image`, `navigator_image` or `virtual_image`. Not sure if I have any favorites.

Reply:
I went with `navigator_signal`